### PR TITLE
feat: add JSON logging

### PR DESCRIPTION
**Pull request by ctdfo**
Looks like this:
image

Also, improved the parsing of the audit event by creating a struct of the JSON instead of unmarshalling it to an interface. See: `internal/eventhub/eventhub_unpacker.go`

### CODE REVIEW

**Review by alexandrebd-dfo — COMMENTED** [review]
A bit confused by the log level setting configuration for the application. It seems we can set the log level for the logger globally. Why not use that and proper info, debug, etc. logging in the rest of the program instead of relying on the config VERBOSELEVEL?

I also see that `flag` is being used in main.go . Will we be passing flags to our binary in the future instead of the current configuration mechanism?

**Review by alexandrebd-dfo — COMMENTED** [review]
Instead of formatting messages with Sprintf, should we use `Msgf` function instead?

File: internal/eventhub/eventhub_runner.go [hunk 1]
```diff
@@ -67,12 +69,14 @@ func Run() {
...
 					// Handle the error, you can log it or take appropriate action
-					fmt.Printf("Error generating random name: %v\n", err)
+					msg := fmt.Sprintf("Error generating random name: %v", err)
+					log.Error().Msg(msg)
 					return
 				}
 
 				if config.VerboseLevel > 1 {
-					fmt.Printf("{%q} > Recieved event pack\n", randomName)
+					msg := fmt.Sprintf("{%v} > Recieved event pack", randomName)
```
Comment (alexandrebd-dfo) [inline internal/eventhub/eventhub_runner.go]: debug level?

File: internal/eventhub/eventhub_unpacker.go [hunk 2]
```diff
@@ -21,30 +22,33 @@ func (h *HubEventUnpacker) InitConfig(f *forwarder.ForwarderConfiguration) {
...
+
+		auditEventStr := record.Properties.Log
 
 		if h.forwarderConfiguration.VerboseLevel > 2 {
-			err := h.ConsoleWriteEventSummary(auditEventStr, mainEventName, i)
-			if err != nil {
-				return err
-			}
+			msg := fmt.Sprintf("%s %d > READ audit event: %s", mainEventName, i, auditEventStr)
+			log.Info().Msg(msg)
```
Comment (alexandrebd-dfo) [inline internal/eventhub/eventhub_unpacker.go]: debug level

**Review by alexandrebd-dfo — COMMENTED** [review]
Instead of formatting messages with Sprintf, should we use `Msgf` function instead?

**Comment by alexandrebd-dfo** [comment]
Overall might want to review the log level of some of the messages. We should be careful to try and avoid highly repetitive messages in `info` logs that do not provide day-to-day valuable info

**Comment by alexandrebd-dfo** [comment]
might want to run `go mod tidy`, not sure that `zerolog` should be in the indirect dependencies

**Comment by ctdfo** [comment]
> A bit confused by the log level setting configuration for the application. It seems we can set the log level for the logger globally. Why not use that and proper info, debug, etc. logging in the rest of the program instead of relying on the config VERBOSELEVEL?
> 
> I also see that `flag` is being used in main.go . Will we be passing flags to our binary in the future instead of the current configuration mechanism?

@alexandrebd-dfo I just followed the verbosity level that was mentioned here:
image

But we could map the levels to the zerolog logging levels (e.g., verbosity 4 is `debug`, 3 is `info`, 2 is `warn` and 1 is `error`. Would you prefer that?

**Comment by alexandrebd-dfo** [comment]
> > A bit confused by the log level setting configuration for the application. It seems we can set the log level for the logger globally. Why not use that and proper info, debug, etc. logging in the rest of the program instead of relying on the config VERBOSELEVEL?
> > I also see that `flag` is being used in main.go . Will we be passing flags to our binary in the future instead of the current configuration mechanism?
> 
> @alexandrebd-dfo I just followed the verbosity level that was mentioned here: image
> 
> But we could map the levels to the zerolog logging levels (e.g., verbosity 4 is `debug`, 3 is `info`, 2 is `warn` and 1 is `error`. Would you prefer that?

why not use the zerolog log levels directly? Just scrap the concept of numbered levels completely and remove it from the code. Instead we can just pass the zerolog log level I think.

**Comment by ctdfo** [comment]
> > > A bit confused by the log level setting configuration for the application. It seems we can set the log level for the logger globally. Why not use that and proper info, debug, etc. logging in the rest of the program instead of relying on the config VERBOSELEVEL?
> > > I also see that `flag` is being used in main.go . Will we be passing flags to our binary in the future instead of the current configuration mechanism?
> > 
> > 
> > @alexandrebd-dfo I just followed the verbosity level that was mentioned here: image
> > But we could map the levels to the zerolog logging levels (e.g., verbosity 4 is `debug`, 3 is `info`, 2 is `warn` and 1 is `error`. Would you prefer that?
> 
> why not use the zerolog log levels directly? Just scrap the concept of numbered levels completely and remove it from the code. Instead we can just pass the zerolog log level I think.

Sou

File: main.go [hunk 3]
```diff
@@ -10,12 +11,28 @@ import (
 )
 
 func main() {
-	// Default level for this example is info, unless debug flag is present
-	debug := flag.Bool("debug", false, "sets log level to debug")
-	zerolog.SetGlobalLevel(zerolog.InfoLevel)
-	if *debug {
-		zerolog.SetGlobalLevel(zerolog.DebugLevel)
+	logLevelStr := strings.ToLower(os.Getenv("LOGLEVEL"))
+	if logLevelStr == "" {
```
Comment (alexandrebd-dfo) [inline main.go]: why not use a default case in the switch later on?

**Review by alexandrebd-dfo — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**